certora/harness/MinterGatewayHarness.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity 0.8.23;
+import "../../src/MinterGateway.sol";
+
+contract MinterGatewayHarness is MinterGateway {
+    constructor(address ttgRegistrar_, address mToken_) MinterGateway(ttgRegistrar_, mToken_) {}
+
+    function totalMSupply() public view returns (uint240) {
+        // Note this is also used to determine the total M supply in
+        // functions implemented in MinterGatway
+        return uint240(IMToken(mToken).totalSupply()); 
+    }
```

Comment:
What is the purpose of this function? Why can't the user of this harness just call `totalSupply` at `mToken` directly?